File: backend/src/merchant/objects/CreateProductRequest.scala
package delivery.merchant.objects

import delivery.shared.objects.{ListingStatus, MerchantId}

final case class CreateProductRequest(
    merchantId: MerchantId,
    name: String,
    description: String,
    price: Double,
    remainingStock: Int,
    listingStatus: ListingStatus
)

File: frontend/src/objects/merchant/apiTypes/CreateProductRequest.ts
import type { ListingStatus, MerchantId } from '@/objects/shared/ids'
import type { ProductBundleGroup, ProductInventoryMode } from '@/objects/merchant/Product'

export interface CreateProductRequest {
  merchantId: MerchantId
  name: string
  description: string
  imageUrl: string
  categoryName: string
  price: number
  remainingStock: number
  listingStatus: ListingStatus
  inventoryMode?: ProductInventoryMode
  maxPerOrder?: number | null
  bundleGroups?: ProductBundleGroup[]
}
